### PR TITLE
Fixed execute documentation

### LINKED ISSUE #3256: Execute docs not rendering some properties correctly
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
Looks like maybe type annotations in properties might be the issue

Screen Shot 2019-10-14 at 4 59 40 AM

### Steps to reproduce the problem
build the docs

### LINKED ISSUE #3257: Assemble docs not formatted correctly
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
Similar to #3256 

Screen Shot 2019-10-14 at 5 00 39 AM

### PR DESCRIPTION
**Pull request by dennis-liu-1**
### Summary

Fixes #3256
Fixes #3257 

### Details and comments

Added a backslash for new line break to fix the first issue. 

The second one is a limitation on sphinx, and the issue is that the colon in ```list[dict{Parameter: Value}]``` is treated non-literally even with a backslash in front of it. (more details see: https://github.com/sphinx-doc/sphinx/issues/4016). Sphinx will do a greedy search on any colon which interferes with formatting on cases such as this. The solution is to treat that whole segment as a code block.

### CODE REVIEW

**Comment by dennis-liu-1** [comment]
On further inspection the second fix fails linting. The two options are to either use the solution above and modify the lint warnings, or simply avoid using the colon in the original parameter altogether.

File: qiskit/execute.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def execute(experiments, backend,
             The delay between experiments will be rep_time.
             Must be from the list provided by the device.
 
-        parameter_binds (list[dict{Parameter: Value}]):
+        parameter_binds (:code:`list[dict{Parameter: Value}]`):
```
Comment (1ucian0) [inline qiskit/execute.py]: This is not a valid type annotation...
```suggestion
        parameter_binds (list[dict{Parameter: Value}]):
```

Comment (dennis-liu-1) [reply qiskit/execute.py]: Yeah that goes back to the main issue of whether or not to include the colon in the parameter type (faulty docstring output vs invalid parameter)

Comment (dennis-liu-1) [reply qiskit/execute.py]: Both solutions are not ideal, but I don't think it's possible to leave the colon in there while using Sphinx at the moment...

Comment (1ucian0) [reply qiskit/execute.py]: Probably, the simplest and clearer way is `list[dict]` and explain in the text what should the structure of `dict` be. Would that fix the issue?

Comment (SooluThomas) [inline qiskit/execute.py]:
```suggestion
        parameter_binds (list[dict]):
```

Comment (dennis-liu-1) [reply qiskit/execute.py]: Looks good to me, also a valid fix

**Pull request merged**